### ISSUE TITLE
[Feature request] Completely customizable bottom generic row

### ISSUE THREAD

**Issue by woj-tek** [AnySoftKeyboard/AnySoftKeyboard#1832]
Currently it's only possible to select bottom row layout from a fixed set of options. It would be great if, instead, it would be possible to edit bottom row by adding/removing desired keys in desired order. I'd imagine it would work like, for example, desktop toolbar editing:
```
[     area to be edited with selected keys      ]
-------------------------------------------------
      [key1] [key2] [key3] [spacebar] etc…
```

Adding/removing would happen by tapping on keys in section; adding new key to layout would locate it at the end of the layout. Most keys would have fixed width and `[spacebar]` would be flexible and would occupy remaining space.

**Comment by rainio** [comment]
That would be very useful as the frequently used apostrophe is accessed through to symbols keys.

**Comment by snkfxx** [comment]
This would be a great feature. It would also be very useful for those using a dvorak layout where '.' and ',' are placed among the letter keys and thus redundant in the bottom row.

**Comment by a-p-fromm** [comment]
Would not it be a possible option to make a package containing costom top and bottom row, so the interested user can edit it the way he/she wants and insert the code this way?

**Comment by woj-tek** [comment]
That would create quite a lot of "plugins" in the app-stores, and most likely they would still be combinations that doesn't fit for particular use cases. In addition, non-technical people wouldn't have possibility to change those…

**Comment by a-p-fromm** [comment]
Sure, you are requesting something comfortable. But you technicaly got one row for search bar, one for adress bar, one for IM and so on. So it is not one but four or five rows. Thouse are changed dynamicaly. Every single button got a lot of options like repeatability, width, pop-up options, long-press and so on... The interface to manage that would be as complex as a xml-script containig all this information. So in the end editing a .xml seems to be easier and smarter way. As well I hardly imagine non-tech people editing their keyboards at all.
What i am talking about is only one, easy to build package in the stores for anyone to start from.
Another option would be external, editable xml-files to build your custom rows. One for top row and one for bottom row.

**Comment by woj-tek** [comment]
> The interface to manage that would be as complex as a xml-script containig all this information.

Not really. And a lot of UIs allow customization. In case of those elements the items and it's properties are fixed and noone wants to change them so we are left with only item/key-ID. UI/UX wise there is abundance of similar concepts (just from the top of my head):
* customization of QuickTiles
* in duolingo - ordering words to make an answer

**Comment by PinkDev1** [comment]
@menny any progress on this? Many of us would really like this feature (this issue has the most "likes" on the repo)
